app/models/product.rb

```diff
@@ -0,0 +1,28 @@
+class Product < ActiveRecord::Base
+  has_and_belongs_to_many :categories
+  belongs_to :seller
+  has_many :order_items
+  has_many :reviews
+
+  # Validations ----------------------------------------------------------------
+  validates :name, presence: true, uniqueness: true
+  validates :price, presence: true, numericality: { greater_than: 0 }
+  validates :seller_id, presence: true, numericality: { only_integer: true }
+  validates :stock, presence: true, numericality: { only_integer: true }
+
+  def self.average_rating(id)
```

Comment:
Why'd you make this a class method? It totes works, I'm just curious as to your reasoning.